src/darwin/Framework/CHIP/MTRDeviceConnectivityMonitor.mm

```diff
@@ -258,7 +258,7 @@ - (void)_stopMonitoring
             dispatch_after(dispatch_time(DISPATCH_TIME_NOW, kSharedConnectionLingerIntervalSeconds * NSEC_PER_SEC), sSharedResolverQueue, ^{
                 std::lock_guard lock(sConnectivityMonitorLock);
 
-                if (!sConnectivityMonitorCount) {
+                if (!sConnectivityMonitorCount && sSharedResolverConnection) {
```

Comment:
i take it this is the "juicy part"?

Reply:
While this is also "juicy", the fix below in `stopMonitoring` is what I'm fixing for a particular crash report. So both are needed.